src/common/lib/client/realtimechannel.ts

```diff
@@ -650,6 +644,7 @@ class RealtimeChannel extends Channel {
         if (!message.presence) break;
       // eslint-disable-next-line no-fallthrough
       case actions.PRESENCE: {
+        this.setChannelSerial(message.channelSerial, false); // RTL15b
```

Comment:
Will this include `channelSerial` from the `SYNC` - as it falls though - which is used for the sync cursor? (which is why I initially put it all at the top as this switch statement is confusing)

Reply:
oooh yes. very good point.

Reply:
https://github.com/ably/ably-js/pull/1058/commits/442f404e56969d1dfe2b18f435a32fa1bfd2264a WDYT? refactors out the fallthrough in the switch statement, makes it a bit more readable

Reply:
Nice looks good - in which case can we split up the commits (to keep refactoring separate from behaviour changes - and remove the original commit)

(sorry I also pushed a change to my branch so this will need a rebase)

Reply:
https://github.com/ably/ably-js/pull/1058/commits/4375b1a4630591064bb3c1a9139d742aa43505fd

Reply:
(with your commits there's no behaviour changes left, all refactoring)